nullaway/src/main/java/com/uber/nullaway/handlers/LibraryModelsHandler.java

```diff
@@ -128,29 +128,32 @@ public boolean onOverrideMayBeNullExpr(
       @Nullable Symbol exprSymbol,
       VisitorState state,
       boolean exprMayBeNull) {
-    if (expr.getKind().equals(Tree.Kind.METHOD_INVOCATION)
-        && exprSymbol instanceof Symbol.MethodSymbol) {
-      OptimizedLibraryModels optLibraryModels = getOptLibraryModels(state.context);
-      Symbol.MethodSymbol methodSymbol = (Symbol.MethodSymbol) exprSymbol;
-      // When looking up library models of annotated code, we match the exact method signature only;
-      // overriding methods in subclasses must be explicitly given their own library model.
-      // When dealing with unannotated code, we default to generality: a model applies to a method
-      // and any of its overriding implementations.
-      // see https://github.com/uber/NullAway/issues/445 for why this is needed.
-      boolean isMethodAnnotated =
-          !getCodeAnnotationInfo(state.context).isSymbolUnannotated(methodSymbol, this.config);
-      if (optLibraryModels.hasNullableReturn(methodSymbol, state.getTypes(), !isMethodAnnotated)
-          || !optLibraryModels.nullImpliesNullParameters(methodSymbol).isEmpty()) {
-        // These mean the method might be null, depending on dataflow and arguments. We force
-        // dataflow to run.
-        return analysis.nullnessFromDataflow(state, expr) || exprMayBeNull;
-      } else if (optLibraryModels.hasNonNullReturn(
-          methodSymbol, state.getTypes(), !isMethodAnnotated)) {
-        // This means the method can't be null, so we return false outright.
+    if (!(expr.getKind() == Tree.Kind.METHOD_INVOCATION
+        && exprSymbol instanceof Symbol.MethodSymbol)) {
+      return exprMayBeNull;
+    }
+    OptimizedLibraryModels optLibraryModels = getOptLibraryModels(state.context);
+    Symbol.MethodSymbol methodSymbol = (Symbol.MethodSymbol) exprSymbol;
+    // When looking up library models of annotated code, we match the exact method signature only;
+    // overriding methods in subclasses must be explicitly given their own library model.
+    // When dealing with unannotated code, we default to generality: a model applies to a method
+    // and any of its overriding implementations.
+    // see https://github.com/uber/NullAway/issues/445 for why this is needed.
+    boolean isMethodUnannotated =
+        getCodeAnnotationInfo(state.context).isSymbolUnannotated(methodSymbol, this.config);
+    if (exprMayBeNull) {
+      if (optLibraryModels.hasNonNullReturn(methodSymbol, state.getTypes(), isMethodUnannotated)) {
```

Comment:
Why not:

```
if (exprMayBeNull) {
    // This is the only case in which we may switch the result from @Nullable to @NonNull: 
    return !optLibraryModels.hasNonNullReturn(methodSymbol, state.getTypes(), isMethodUnannotated);
}
```

Guess there is some consistency in all returns being boolean literals, but not sure the extra if nesting is worth it.

Reply:
i am aware of this simplification, i have mentioned it here before:

https://github.com/uber/NullAway/pull/747#discussion_r1133245711

> when there is only a single reason, we could of course use the condition in a single return statement if preferred?

guess back then there was no preference for the single return statement, but happy to just it now here

the suggested comment is true now, but will it stay true in the future as well?
or is this comment really just local to the `LibraryModelsHandler` logic?

Reply:
let me know your preference and whether we should also adjust the other `onOverrideMayBeNullExpr` implementations that also only have a single `if` (for consistency?)

Reply:
I just think in this case it's clearer to have the return with the call just to avoid extra if nesting and thus the issue of matching branch conditions to returns, _but_ that's a personal preference. Don't feel super strongly either way. In the linked example, I agree that having each condition in its own if check is more readable than some complicated boolean expression with `||`/`&&`, but this case here is a single call and everything else is nested just one level deep.